core/services/relay/evm/event_binding.go

```diff
@@ -326,6 +316,91 @@ func (e *eventBinding) decodeLog(ctx context.Context, log *logpoller.Log, into a
 	return mapstructureDecode(topicsInto, into)
 }
 
+func (e *eventBinding) decodeLogsIntoSequences(ctx context.Context, logs []logpoller.Log, into any) ([]commontypes.Sequence, error) {
+	sequences := make([]commontypes.Sequence, len(logs))
+
+	for idx := range logs {
+		sequences[idx] = commontypes.Sequence{
+			Cursor: fmt.Sprintf("%s-%s-%d", logs[idx].BlockHash, logs[idx].TxHash, logs[idx].LogIndex),
+			Head: commontypes.Head{
+				Identifier: fmt.Sprint(logs[idx].BlockNumber),
+				Hash:       logs[idx].BlockHash.Bytes(),
+				Timestamp:  uint64(logs[idx].BlockTimestamp.Unix()),
+			},
+		}
+
+		var tpVal reflect.Value
+
+		tpInto := reflect.TypeOf(into)
```

Comment:
this would help with readability a bit
```suggestion

		var typeVal reflect.Value

		typeInto := reflect.TypeOf(into)
```